src/app/cli/src/app.rs

```diff
@@ -135,19 +135,38 @@ pub async fn run(
         cli_catalog.get_one::<GcService>()?.evict_cache()?;
     }
 
-    let result = match cli_commands::get_command(&base_catalog, &cli_catalog, &matches) {
-        Ok(mut command) => {
-            if command.needs_workspace() && !workspace_svc.is_in_workspace() {
-                Err(CLIError::usage_error_from(NotInWorkspace))
-            } else if command.needs_workspace() && workspace_svc.is_upgrade_needed()? {
-                Err(CLIError::usage_error_from(WorkspaceUpgradeRequired))
-            } else if current_account.is_explicit() && !is_multi_tenant_workspace {
-                Err(CLIError::usage_error_from(NotInMultiTenantWorkspace))
-            } else {
-                command.run().await
+    // TODO: Extend Command trait with the corresponding property
+    let need_to_wrap_with_transaction = matches!(
+        matches.subcommand(),
+        Some(("login" | "add" | "delete" | "pull", _))
+    );
```

Comment:
I think there is nothing to wait for, do this extension

Reply:
Unfortunately, since we need to know if there will be a transaction (!) even before a command is created, I had to do it with a separate function `cli_commands::command_needs_transaction()`